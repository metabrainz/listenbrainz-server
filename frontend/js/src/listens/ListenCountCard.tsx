import * as React from "react";
import Card from "../components/Card";
import GlobalAppContext from "../utils/GlobalAppContext";

export type ListenCountCardProps = {
  listenCount?: number;
  user: ListenBrainzUser;
};

function ListenCountCard(props: ListenCountCardProps) {
  const { currentUser } = React.useContext(GlobalAppContext);
  const { listenCount, user } = props;
  const isCurrentUser = currentUser?.name === user?.name;

  let content;

  if (listenCount) {
    content = (
      <div>
        {isCurrentUser ? "You have" : `${user.name} has`} listened to
        <hr />
        {listenCount.toLocaleString()}
        <br />
        <small className="text-muted">songs so far</small>
      </div>
    );
  } else {
    content = (
      <>
        <p className="text-muted">
          {isCurrentUser ? "Your" : `${user.name}'s`} listens count
        </p>
        <hr style={{ margin: "10px 0px" }} />
        <div style={{ fontSize: "14px" }} className="text-muted">
          {isCurrentUser ? "You haven't" : `${user.name} hasn't`} listened to
          any songs yet.
        </div>
      </>
    );
  }

  return (
    <Card id="listen-count-card" data-testid="listen-count-card">
      {content}
    </Card>
  );
}

export default ListenCountCard;
