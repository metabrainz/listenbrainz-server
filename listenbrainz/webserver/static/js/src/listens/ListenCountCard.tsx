import * as React from "react";
import Card from "../components/Card";

export type ListenCountCardProps = {
  listenCount?: number;
};

const ListenCountCard = (props: ListenCountCardProps) => {
  const { listenCount } = props;

  return (
    <Card id="listen-count-card">
      <h4>
        You have listened to
        <hr />
      </h4>
      <div>
        <h4>{listenCount}</h4>
        <small className="text-muted">songs so far</small>
      </div>
    </Card>
  );
};

export default ListenCountCard;
