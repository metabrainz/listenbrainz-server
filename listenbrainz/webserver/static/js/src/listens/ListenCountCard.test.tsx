import * as React from "react";
import { mount } from "enzyme";

import ListenCountCard, { ListenCountCardProps } from "./ListenCountCard";

const props: ListenCountCardProps = {
  listenCount: 100,
};

describe("ListenCountCard", () => {
  it("renders correctly", () => {
    const wrapper = mount(<ListenCountCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
