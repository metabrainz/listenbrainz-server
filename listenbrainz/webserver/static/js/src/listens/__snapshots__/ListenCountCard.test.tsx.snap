// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListenCountCard renders correctly 1`] = `
<ListenCountCard
  listenCount={100}
>
  <ForwardRef
    id="listen-count-card"
  >
    <div
      id="listen-count-card"
      style={
        Object {
          "background": "#FFFFFF",
          "border": "1px solid #EEEEEE",
          "borderRadius": "8px",
          "boxShadow": "0 1px 1px rgba(192,192,192,0.1), 0 2px 2px rgba(192,192,192,0.15), 0 4px 4px rgba(192,192,192,0.20)",
          "boxSizing": "border-box",
          "height": "100%",
        }
      }
    >
      <h4>
        You have listened to
        <hr />
      </h4>
      <div>
        <h4>
          100
        </h4>
        <small
          className="text-muted"
        >
          songs so far
        </small>
      </div>
    </div>
  </ForwardRef>
</ListenCountCard>
`;
